ioctl/test_ioctl.c

```diff
@@ -0,0 +1,405 @@
+#define DEV      "/dev/test"
+/* In above T stands for test, do not confuse with tty */
+#define TIOCSIG       _IOC(IOC_VOID, 'T', 0x01, 0)
+#define TIOCVAL       _IOC(IOC_VOID, 'T', 0x02, 4)
+#define TIOCPTR       _IOC(IOC_IN, 'T', 0x03, 4)
+#define TIOCPTR_BIG   _IOC(IOC_IN, 'T', 0x04, BIG_SIZE)
+#define TIOCOUT       _IOC(IOC_OUT, 'T', 0x05, 4)
+#define TIOCOUT_BIG   _IOC(IOC_OUT, 'T', 0x06, BIG_SIZE)
+#define TIOCINOUT     _IOC(IOC_INOUT, 'T', 0x07, 4)
+#define TIOCINOUT_BIG _IOC(IOC_INOUT, 'T', 0x08, BIG_SIZE)
+
+
+static void *thread(void *arg)
+{
+	msg_t msg;
+	unsigned long int rid;
+	int32_t flag = 1;
+	int32_t out;
+	int8_t out2[BIG_SIZE];
+	void *retval = &ret;
+
+	while (loop_flag) {
+		if (msgRecv(port, &msg, &rid) < 0) {
+			continue;
+		}
+
+		if (msg.type == mtDevCtl) {
+			unsigned long request;
+			int err = 0;
+
+			const void *out_data = NULL;
+			const void *in_data = ioctl_unpack(&msg, &request, NULL);
+
+			switch (request) {
+				case TIOCVAL:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCVAL request");
+					}
+					if (write(fd, &in_data, 4) != 4) {
+						TEST_MESSAGE("write failed in TIOCVAL request");
+					}
+					break;
+				case TIOCSIG:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCSIG request");
+					}
+					if (write(fd, &flag, 4) != 4) {
+						TEST_MESSAGE("write failed in TIOCSIG request");
+					}
+					break;
+				case TIOCPTR:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCPTR request");
+					}
+					if (write(fd, in_data, 4) != 4) {
+						TEST_MESSAGE("write failed in TIOCPTR request");
+					}
+					break;
+				case TIOCPTR_BIG:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCPTR_BIG request");
+					}
+					if (write(fd, in_data, BIG_SIZE) != BIG_SIZE) {
+						TEST_MESSAGE("write failed in TIOCPTR_BIG request");
+					}
+					break;
+				case TIOCOUT:
+					out = 15;
+					out_data = (const void *)(&out);
+					break;
+				case TIOCOUT_BIG:
+					memset(out2, 5, BIG_SIZE);
+					out_data = (const void *)(out2);
+					break;
+				case TIOCINOUT:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCINOUT request");
+					}
+					if (write(fd, in_data, 4) != 4) {
+						TEST_MESSAGE("write failed in TIOCINOUT request");
+					}
+					out = 18;
+					out_data = (const void *)(&out);
+					break;
+				case TIOCINOUT_BIG:
+					if (lseek(fd, 0, SEEK_SET) != 0) {
+						TEST_MESSAGE("lseek failed in TIOCINOUT_BIG request");
+					}
+					if (write(fd, in_data, BIG_SIZE) != BIG_SIZE) {
+						TEST_MESSAGE("write failed in TIOCINOUT_BIG request");
+					}
+					memset(out2, 8, BIG_SIZE);
+					out_data = (const void *)(out2);
+					break;
+				default:
+					err = -1;
+					break;
+			}
+			ioctl_setResponse(&msg, request, err, out_data);
+		}
+		msgRespond(port, &msg, rid);
+	}
+	portDestroy(port);
+	return retval;
```

Comment:
result not used, remove?

Reply:
Not used but thread function is declared as such that returns void*
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/108683715/2050c7d5-3958-4a63-902f-fb2be9ad7d19)

Reply:
just `return NULL`